scrapy/utils/defer.py

```diff
@@ -114,3 +118,25 @@ def iter_errback(iterable, errback, *a, **kw):
             break
         except Exception:
             errback(failure.Failure(), *a, **kw)
+
+
+def _isfuture(o):
+    # workaround for Python before 3.5.3 not having asyncio.isfuture
+    if hasattr(asyncio, 'isfuture'):
+        return asyncio.isfuture(o)
+    return isinstance(o, asyncio.Future)
```

Comment:
If the coverage data is correct, `_isfuture` could be replaced by `asyncio.isfuture` and work the same. I wonder what the best approach here would be. Maybe we should just use `asyncio.isfuture`.

If we need to exclude future-like content that are not `Future` instances, such as `Task` instances, I think we may need a more complex check.

Reply:
We don't test on Python before 3.5.3 :)

Reply:
:man_facepalming: 

:thinking: I think we should either change that or not bother supporting 3.5.3-.

Reply:
Yup.
My only concern is that the current Python3 version in Ubuntu 16.04 LTS is 3.5.2. Though I couldn't find how to make travis actually use the system Python3.

Reply:
Let's see if it works fine on travis 3.5.2.

Reply:
The proxy_connect tests will fail. In https://github.com/scrapy/scrapy/pull/4261 I’ve installed 3.5.0, and it works, but I’ve had to skip mitmproxy-based tests, as for Python 3.5.3 and earlier you need to use mitmproxy 2, and even after updating the command-line parameters accordingly I got timeouts.

Reply:
Yet https://travis-ci.org/scrapy/scrapy/jobs/631286391 worked with 3.5.2

Reply:
Hmm… It worked…